### PR TITLE
Allow multiple symbols for any state of a checkbox

### PR DESCRIPTION
**Pull request by deepredsky**
This patch allows checkboxes to have multiple symbols that can denote a state

For example:

```
   let g:vimwiki_list = [{'path': '~/path/', 'listsyms': [' ', '.', 'o', 'O', ['x', 'X']] }]
```

This will allow the completed checkbox to be represented by both 'x' and 'X'

Steps for submitting a pull request:

- [x] **ALL** pull requests should be made against the `dev` branch!
- [ ] Take a look at [CONTRIBUTING.MD](https://github.com/vimwiki/vimwiki/blob/dev/CONTRIBUTING.md)
- [ ] Reference any related issues.
- [ ] Provide a description of the proposed changes.
- [ ] PRs must pass Vint tests and add new Vader tests as applicable.
- [ ] Make sure to update the documentation in `doc/vimwiki.txt` if applicable,
      including the Changelog and Contributors sections.

### CODE REVIEW

**Comment by tinmarino** [comment]
Hi @deepredsky thank you for the try: "Nothing worth the force of a volunter" (top gun) and "the travel of 1000 miles begins with one step (Lao Tzi).

Good: I love the api (perfect and retrocompatible):    `let g:vimwiki_listsyms = ['', 'o', 'O', ['x', 'X']]`
Bad: Some test are failing, 
```
E716: Key not present in Dictionary: listsyms_list
Undefined variable: v:t_list
```

I think you are still making changes so I will have a look later. Don't hesitate to ping me (even multiple times).

File: autoload/vimwiki/lst.vim [hunk 1]
```diff
@@ -865,6 +869,11 @@ function! s:rate_to_state(rate) abort
     let index = float2nr(ceil(a:rate/100.0*(n-2)))
     let state = listsyms_list[index]
   endif
+
+  if type(state) == v:t_list
```
Comment (tinmarino) [inline autoload/vimwiki/lst.vim:873]: replace with
```vim
type(state) == type([])
```

File: autoload/vimwiki/vars.vim [hunk 2]
```diff
@@ -514,7 +514,7 @@ function! s:get_default_wikilocal() abort
         \ 'list_ignore_newline': {'type': type(0), 'default': 1, 'min': 0, 'max': 1},
         \ 'list_margin': {'type': type(0), 'default': -1, 'min': -1},
         \ 'listsym_rejected': {'type': type(''), 'default': vimwiki#vars#get_global('listsym_rejected')},
-        \ 'listsyms': {'type': type(''), 'default': vimwiki#vars#get_global('listsyms')},
```
Comment (tinmarino) [inline autoload/vimwiki/vars.vim:517]: We should accept both symbols to be rectrocompatible: vimwiki_listsyms can be a string or a list

File: autoload/vimwiki/lst.vim [hunk 3]
```diff
@@ -772,8 +773,11 @@ function! s:get_rate(item) abort
   if state == vimwiki#vars#get_global('listsym_rejected')
     return -1
   endif
-  let n = len(vimwiki#vars#get_wikilocal('listsyms_list'))
```
Comment (tinmarino) [inline autoload/vimwiki/lst.vim:775]: listsyms_list -> listsyms otherwise: E716: Key not present in Dictionary: listsyms_list

File: autoload/vimwiki/vars.vim [hunk 4]
```diff
@@ -1071,34 +1071,41 @@ function! s:populate_list_vars(wiki) abort
...
   " Guard: Check if listym_rejected is in listsyms
-  if match(a:wiki.listsyms, '[' . a:wiki.listsym_rejected . ']') != -1
+  if index(a:wiki.listsyms, a:wiki.listsym_rejected) != -1
     call vimwiki#u#warn('the value of listsym_rejected ('''
           \ . a:wiki.listsym_rejected . ''') must not be a part of listsyms ('''
-          \ . a:wiki.listsyms . ''')')
+          \ . join(a:wiki.listsyms, '') . ''')')
   endif
 
+  if type(a:wiki.listsyms_list[-1]) == v:t_list
```
Comment (tinmarino) [inline autoload/vimwiki/vars.vim:1083]: == type([])

File: autoload/vimwiki/vars.vim (hunk 2 above)
Comment (deepredsky) [reply autoload/vimwiki/vars.vim:517]: is there a way i can specify union type here?

**Comment by deepredsky** [comment]
>  Bad: Some test are failing

I've managed to fix tests in 8* and neovim version. Tests in 7* are still failing and there are some edge case I've found when manually testing. I might need some help with 7* tests :)

**Comment by tinmarino** [comment]
Yes, bad luck for the Vim7.3 error, I did not find where it comes from (`Vim(let):E121: Undefined variable: _`). Others like myself often have some problem with old Vim compatibility. In that case it seems to be in an eval, some dhicotomic print (or `;debug`)could find the place.

Do not spend too much time on this boring topic.

In the other hand, maybe you could add some non-regression tests specific to this feature: it'll be easyer to integrate.

**Comment by deepredsky** [comment]
> Yes, bad luck for the Vim7.3 error, I did not find where it comes from (`Vim(let):E121: Undefined variable: _`). Others like myself often have some problem with old Vim compatibility. In that case it seems to be in an eval, some dhicotomic print (or `;debug`)could find the place.
> 
> Do not spend too much time on this boring topic.
> 
> In the other hand, maybe you could add some non-regression tests specific to this feature: it'll be easyer to integrate.

i fixed all the tests as well, for reason travis did not trigger in my last commit, they do pass locally though.

**Comment by deepredsky** [comment]
stale PR, closing it for now.
